source/actions3/AMActionRunner3.cpp

```diff
@@ -1045,4 +1085,30 @@ bool AMActionRunner3::runActionImmediatelyInQueue(AMAction3 *action)
 	return true;
 }
 
+bool AMActionRunner3::isActionRunnerPausable() const
+{
+	bool isPausable = isActionRunnerPauseEnabled_;
+
+	if (currentAction_) {
+		AMAction3::State currentActionState = currentAction_->state();
+
+		isPausable = isPausable && currentAction_->canPause() && (currentActionState == AMAction3::Running || currentActionState == AMAction3::Paused);
```

Comment:
I think so. This code is originally from the `void AMActionRunnerCurrentViewBase::onStateChanged(int state, int previousState)`. 

I was thinking that it is not necessary. But I keep it just in case I missed something